fe/fe-core/src/main/java/com/starrocks/common/util/concurrent/lock/Locker.java

```diff
@@ -208,6 +209,26 @@ public void unlockDatabases(List<Database> dbs, LockType lockType) {
         }
     }
 
+    /**
+     * Try to lock databases in ascending order of id.
+     * @return: true if all databases are locked successfully, false otherwise.
+     */
+    public boolean tryLockDatabases(List<Database> dbs, LockType lockType, long timeout, TimeUnit unit) {
+        if (dbs == null) {
+            return false;
+        }
+        dbs.sort(Comparator.comparingLong(Database::getId));
+        List<Database> lockedDbs = Lists.newArrayList();
+        for (Database db : dbs) {
+            if (!tryLockDatabase(db, lockType, timeout, unit)) {
+                lockedDbs.stream().forEach(t -> unLockDatabase(t, lockType));
```

Comment:
It's better to put the unlocking code in a try/finally block so that the lock is always released.